### PR TITLE
En_Tanron5

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c [hunk 1]
```diff
@@ -28,32 +35,493 @@ const ActorInit En_Tanron5_InitVars = {
...
+    /* 0x0C */ f32 unk_0C;
+    /* 0x10 */ f32 unk_10;
+    /* 0x14 */ f32 unk_14;
+    /* 0x18 */ Vec3f unk_18;
+    /* 0x24 */ u8 unk_24;
+    /* 0x26 */ s16 unk_26;
+    /* 0x28 */ char unk28[0x4];
+    /* 0x2C */ s16 unk_2C;
+    /* 0x2E */ char unk2E[0x6];
+    /* 0x34 */ f32 unk_34;
```
Comment (jpburnett) [inline src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:91]:
```suggestion
    /* 0x26 */ s16 unk_26;
    /* 0x28 */ UNK_TYPE1 unk28[0x4];
    /* 0x2C */ s16 unk_2C;
    /* 0x2E */ UNK_TYPE1 unk2E[0x6];
    /* 0x34 */ f32 unk_34;
```

Could these pads be removed at all?

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c [hunk 2]
```diff
@@ -28,32 +35,493 @@ const ActorInit En_Tanron5_InitVars = {
...
+
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void func_80BE5818(Actor* thisx, GlobalContext* globalCtx2) {
+    f32 sp6C;
+    s32 i;
+    Vec3f sp5C;
+    EnTanron5* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
```
Comment (jpburnett) [inline src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:402]:
```suggestion
    GlobalContext* globalCtx = globalCtx2;
    EnTanron5* this = THIS;
    f32 sp6C;
    s32 i;
    Vec3f sp5C;
```
Would this still match?

**Review by jpburnett — COMMENTED** [review]
nice work, minor comments on padding and the ordering of stack variables.

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c (hunk 2 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.c:402]: Nope, it doesn't. I had it like this but I have to move them down for stack.

File: src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h [hunk 3]
```diff
@@ -5,9 +5,26 @@
...
 
+#define ENTANRON5_GET(thisx) ((thisx)->params)
+
+enum {
+    /* 000 */ ENTANRON5_0,
+    /* 100 */ ENTANRON5_100 = 100,
+    /* 107 */ ENTANRON5_107 = 107,
+    /* 108 */ ENTANRON5_108 = 108,
+    /* 110 */ ENTANRON5_110 = 110,
+};
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Tanron5/z_en_tanron5.h:16]: This looks like these would be better as defines

File: include/z64math.h [hunk 4]
```diff
@@ -5,6 +5,11 @@
 
 #define VEC_SET(V,X,Y,Z) V.x=X;V.y=Y;V.z=Z
 
+typedef struct {
+    /* 0x00 */ s16 x;
+    /* 0x02 */ s16 z;
+} Vec2s; // size = 0x04
+
```
[the remark this answers is not in the document]
Comment (jpburnett) [reply include/z64math.h:12]: I looked at z64math.h and saw a `Vec2f`, so I figured the existence of a Vec2s seemed plausible.

[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply include/z64math.h:12]: It seems like the sort of type that could be more widely used. Perhaps put a label on it and check it later.

**Review by EllipticEllipsis — APPROVED** [review]

File: include/z64math.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (hensldm) [reply include/z64math.h:12]: I'm pretty I've seen other `Vec2s` defines in actors, though can't seem to find them anymore. Personally I think it makes sense here with `Vec2f`, though I would prefer for its members to match `Vec2f`, either `x and y` or `x and z`.

**Review by hensldm — APPROVED** [review]

File: include/z64math.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (Kelebek1) [reply include/z64math.h:12]: I changed everything to use x/z.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**